### PR TITLE
Allow converted nulls in expr-tree coalesce

### LINKED ISSUE #4593: [Regression] C# rejects (int?)null ?? (int?)5 in an expression tree
Previous C# compilers allowed the following code, but the Roslyn compiler rejects it:

```cs
using System;
using System.Linq;
using System.Linq.Expressions;

class Program
{
    static void Main(string[] args)
    {
      var query = Enumerable.Empty<object>().AsQueryable().Select(a => (int?)null ?? (int?)5);
    }
}
```

### PR DESCRIPTION
**Pull request by TyOverby**
Closes #4593

This fixes a regression in the Expression Tree diagnostic pass.

In expression trees, the native compiler would ban `null ?? new object()`, but allow `(int?) null ?? (int?) 5`.

This moves us back to the old behavior.

### CODE REVIEW

**Comment by TyOverby** [comment]
CC: @dotnet/roslyn-compiler 

@VSadov: I will add tests for expression-tree structure and ping you before I merge.

File: src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs [hunk 1]
```diff
@@ -546,7 +546,7 @@ public override BoundNode VisitNameOfOperator(BoundNameOfOperator node)
 
         public override BoundNode VisitNullCoalescingOperator(BoundNullCoalescingOperator node)
         {
-            if (_inExpressionLambda && node.LeftOperand.ConstantValue != null && node.LeftOperand.ConstantValue.IsNull)
+            if (_inExpressionLambda && node.LeftOperand.IsLiteralNull())
```
Comment (TyOverby) [inline src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs]: Write a comment here about which kinds of null are allowed.

**Comment by AlekseyTs** [comment]
LGTM, modulo a test for the expression tree shape.

**Comment by VSadov** [comment]
LGTM. Needs ET test for the positive scenario

**Comment by jaredpar** [comment]
:+1: modulo adding a test for positive case.

**Comment by TyOverby** [comment]
I added a test that checks to make sure that the expression tree is constructed correctly.  I based this off of a decompiled assembly from the native compiler.

**Comment by TyOverby** [comment]
Ping @VSadov @AlekseyTs @jaredpar.

That extra test has been added.  If it looks good to you, I'll merge now.

**Comment by VSadov** [comment]
LGTM

**Pull request merged**